USER_GUIDE.md

```diff
@@ -68,6 +68,8 @@ Transport transport = new RestClientTransport(restClient,
 OpenSearchClient client = new OpenSearchClient(transport);
 ```
 
+Upcoming OpenSearch `3.0.0` release brings HTTP/2 support and as such, the `RestClientTransport` would switch to HTTP/2 if available (for both HTTPS and/or HTTP protocols). The desired protocol could be forced using `RestClientBuilder.HttpClientConfigCallback`.
```

Comment:
How can I see what protocol has been chosen?
Example of how I would switch to/from HTTP/2?

Reply:
> How can I see what protocol has been chosen?

Basically this could be done by:
 - inspecting the traffic
 - logging requests and responses (they contain HTTP protocol version)

I am not sure what kind of prove we could provide here. 

>  Example of how I would switch to/from HTTP/2?
`RestClientBuilder.HttpClientConfigCallback` should be documented in core but I will add the configuration snippet for Apache HttpClient 5 transport.

Reply:
For documentation generally I ask myself "what would I do". So if you think logging is how you'd do it, I'd show here (or somewhere in the docs) how to log the HTTP version chosen.

Reply:
> For documentation generally I ask myself "what would I do". So if you think logging is how you'd do it, I'd show here (or somewhere in the docs) how to log the HTTP version chosen.

This is not something we should be worried about here (I think), there are few things involved on our behalf:
 - for HTTPS, we have the ALPN protocol in place 
 - for HTTP, we have `Upgrade` in place

The OpenSearch Java client could connect to `2.x` clusters fe, that would mean using HTTP/1.1 no matter what. Generally, forcing the protocol is the only reliable way (I am aware of) to make sure only HTTP/1.1 or only HTTP/2 has to be used. 

To this point, we would be building features on top of HTTP/2 very soon (I believe), so the presence of HTTP/2 would be obvious.

Reply:
Fair enough.